lib/puppet/functions/foreman_proxy/generate_uuid.rb

```diff
@@ -0,0 +1,9 @@
+Puppet::Functions.create_function(:'foreman_proxy::generate_uuid') do
+  dispatch :uuid do
+    # no arguments
```

Comment:
Might be nice to use document the return type
```suggestion
    return_type 'String'
```